src/main/java/org/ambraproject/wombat/model/RelatedArticle.java

```diff
@@ -21,50 +21,113 @@
+          .build();
+    }
+}
+  public abstract String getDoi();
+  public abstract LocalDate getPublicationDate();
+  public abstract Integer getRevisionNumber();
+  public abstract String getTitle();
+  public abstract String getType();
+  public abstract RelatedArticleJournal getJournal();
 
-  public LocalDate getPublicationDate() {
-    return publicationDate;
+  public boolean isPublished() {
+    return this.getRevisionNumber() != null;
   }
 
-  public RelatedArticle(String doi, String title, LocalDate publicationDate) {
-    this.doi = doi;
-    this.title = title;
-    this.publicationDate = publicationDate;
+  public static Builder builder() {
+    return new AutoValue_RelatedArticle.Builder();
   }
 
-  @Override
-  public boolean equals(Object o) {
-    if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
+  public static Map<String, String> displayType = ImmutableMap.<String, String>builder()
+    .put("commentary", "Commentary")
+    .put("commentary-article", "Commentary")
+    .put("companion", "Companion")
+    .put("corrected-article", "Correction")
+    .put("correction-forward", "Correction")
+    .put("retracted-article", "Retraction")
+    .put("retraction-forward", "Retraction")
+    .put("object-of-concern", "Expression of Concern")
+    .put("concern-forward", "Expression of Concern")
+    .put("updated-article", "Update Article")
+    .put("update-forward", "Update Article")
+    .build();
 
-    RelatedArticle that = (RelatedArticle) o;
+  public static Map<String, Integer> typeDisplayOrder = ImmutableMap.<String, Integer>builder()
+    .put("Retraction", 1)
+    .put("Correction", 2)
+    .put("Expression of Concern", 3)
+    .put("Update Article", 4)
+    .put("Companion", 6)
+    .build();
 
-    if (!doi.equals(that.doi)) return false;
-    if (!title.equals(that.title)) return false;
-    return publicationDate.equals(that.publicationDate);
+  public static Comparator<String> SORT_BY_TYPE_COMPARATOR =
+    Comparator.comparing((str)->typeDisplayOrder.getOrDefault(str, 100));
 
+  public String getDisplayType() {
+    return displayType.getOrDefault(getType(), getType());
   }
 
-  @Override
-  public int hashCode() {
-    int result = doi.hashCode();
-    result = 31 * result + title.hashCode();
-    result = 31 * result + publicationDate.hashCode();
-    return result;
+  /*
+    * Return true if the related article here is a "has" type-relation
+    * to the article it is attached to. In other words, if the base
+    * article can be said to "has correction" of this article. Used
+    * for display.
+   */
+  public boolean isHasRelation() {
+    String type = getType();
+    return (type.equals("companion") ||
+            type.equals("corrected-article") ||
+            type.equals("object-of-concern") ||
+            type.equals("retracted-article") ||
+            type.equals("updated-article") ||
+            type.equals("commentary-article"));
+  }
```

Comment:
Wombat doesn't connect to a database. :)

Even if it did, I might push back a bit on this, because I think this is purely display related, and our database controls nothing about the display of our content at the moment. So I'd be a little concerned about starting to use bits and pieces of the database to control certain display capabilities, e.g. sort order for related article types.

Reply:
well, I meant to put this in Rhino's database, then to get this info through Rhino.

ok, I hear ya.